mdp/nodes/recursive_expansion_nodes.py

```diff
@@ -0,0 +1,623 @@
+"""
+This module provides nodes for recursive computation of sequences of functions.
+"""
+import mdp
+from mdp import numx as np
+from mdp.nodes import PolynomialExpansionNode
+__docformat__ = "restructuredtext en"
+
+
+def recf_standard_poly(result, x, special, n, cur_var, pos):
+    """Implementation of the recursion formula for standard polynomials.
+    The recursion formula is P_n = P_{n-1} * x.
+
+    :param result: Contains the observations along the first dimension
+        and the different function values of expansion w.r.t. an observation
+        along the second dimension.
+    :type result: numpy.ndarray
+
+    :param x: The data to be expanded.
+    :type x: numpy.ndarray
+
+    :param special: Index of a special element to be considered in the
+        recursion formula. E.g. the first order function value.
+    :type special: int
+
+    :param n: The order of the function to be computed at this step.
+    :type n: int
+
+    :param cur_var: The index of the current variable to be considered in the
+        recursion. This value will have to be lower than x.shape[1].
+    :type cur_var: int
+
+    :param pos: The index of the element to be computed, along the second
+        dimension of result.
+    :type int: int
+
+    :returns: The vectorized result (along the observations) of the
+        n-th recursion step of the cur_var-th variable.
+    :rtype: numpy.ndarray
+    """
+    return result[:, pos-1]*result[:, special]
+
+
+def init_standard_poly(result, x, pos, cur_var):
+    """Initialize the first order before starting the recursion.
+
+    Mathematically this amounts to
+    P_1 = x.
+
+    :param result: Contains the observations along the first dimension
+        and the different function values of expansion w.r.t. an observation
+        along the second dimension.
+    :type result: numpy.ndarray
+
+    :param x: The data to be expanded.
+    :type x: numpy.ndarray
+
+    :param pos: The index of the element to be computed, along the second
+        dimension of result.
+    :type int: int
+
+    :param cur_var: The index of the current variable to be considered in the
+        recursion. This value will have to be lower than x.shape[1].
+    :type cur_var: int
+
+    :returns: The index of the next element that will be computed in the
+        recursion, the order of the next element, and the index of a special
+        member.
+    :rtype: numpy.ndarray
```

Comment:
The return type is actually (int, int, int), or better as PEP 484 Tuple[int, int, int]?
Same in doc of the other init functions.

Reply:
Fixed.